src/hal/tiva_c/spi.rs

```diff
@@ -0,0 +1,304 @@
+// Zinc, the bare metal stack for rust.
+// Copyright 2014 Lionel Flandrin <lionel@svkt.org>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! SPI configuration
+
+/// Enables SPI interface on any of the 4 SSI (Synchronous Serial Interface) modules in TM4C microcontrollers
+
+use core::intrinsics::abort;
+use hal::tiva_c::sysctl;
+use util::support::get_reg_ref;
+
+#[path="../../util/ioreg.rs"]
+#[macro_use] mod ioreg;
+#[path="../../util/wait_for.rs"]
+#[macro_use] mod wait_for;
+
+/// There are 4 SSI instances an SPI interface can use
+/// See the TM4C123GH6PM datasheet page 954 for detailed signal to pin mappings
+#[allow(missing_docs)]
+#[derive(Clone, Copy)]
+pub enum SpiId {
+  /// SPI over SSI0, uses pins PA2:5
+  Spi0,
+
+  /// SPI over SSI1, uses pins PF0:3 or PD0:3
+  Spi1,
+
+  /// SPI over SSI2, uses pins PB4:7
+  Spi2,
+
+  /// SPI over SSI3, uses pins PD0:3
+  Spi3,
+}
+
+/**
+SPI configuration object
+
+Configuration object used when instantiating an SPI instance
+
+Note: The SPI GPIO pins must be set up correctly before using SPI. For example, for a master-only TX and CK config on SSI0,
```

Comment:
Can you change the note into TODO? This actually belongs to platformtree macro, even if it's broken now.